code/modules/unit_tests/lungs.dm

```diff
@@ -0,0 +1,145 @@
+#define LUNGS_TEST_RETURN(breath_ok, test_lungs) if(!breath_ok)\
+	TEST_FAIL("[test_lungs.type]/check_breath() returned falsy / status code 0 (failure) when it wasn't expected.")\
+
+#define LUNGS_TEST_RETURN_FAIL(breath_ok, test_lungs) if(breath_ok)\
+	TEST_FAIL("[test_lungs.type]/check_breath() returned truthy / status code 1 (success) when it wasn't expected.")\
+
+#define LUNGS_TEST_BREATHE(test_lungs, lab_rat, needed_gases) if(lab_rat.failed_last_breath)\
+	TEST_FAIL("[test_lungs.type]/check_breath() can't get a full breath from [needed_gases].")\
+
+#define LUNGS_TEST_SUFFOCATE(test_lungs, lab_rat, suffocant) if(!lab_rat.failed_last_breath)\
+	TEST_FAIL("[test_lungs.type]/check_breath() didn't suffocate from [suffocant] when expected.")\
+
+/// Ensures the correct gas alerts are displayed after Lungs take a breath.
+/datum/unit_test/lungs_sanity/proc/check_gas_alerts(mob/living/carbon/human/lab_rat, obj/item/organ/internal/lungs/test_lungs, datum/gas_mixture/test_mix)
+	// Check "not enough" gas alerts.
+	if(lab_rat.has_alert(ALERT_NOT_ENOUGH_OXYGEN) && (!test_lungs.safe_oxygen_min || test_mix.has_gas(/datum/gas/oxygen)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_NOT_ENOUGH_OXYGEN when it wasn't expeced.")
+	if(lab_rat.has_alert(ALERT_NOT_ENOUGH_NITRO) && (!test_lungs.safe_nitro_min || test_mix.has_gas(/datum/gas/nitrogen)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_NOT_ENOUGH_NITROGEN when it wasn't expeced.")
+	if(lab_rat.has_alert(ALERT_NOT_ENOUGH_CO2) && (!test_lungs.safe_co2_min || test_mix.has_gas(/datum/gas/carbon_dioxide)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_NOT_ENOUGH_CO2 when it wasn't expeced.")
+	if(lab_rat.has_alert(ALERT_NOT_ENOUGH_PLASMA) && (!test_lungs.safe_plasma_min || test_mix.has_gas(/datum/gas/plasma)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_NOT_ENOUGH_PLASMA when it wasn't expeced.")
+	// Check "too much" gas alerts.
+	if(lab_rat.has_alert(ALERT_TOO_MUCH_OXYGEN) && (!test_lungs.safe_oxygen_max || !test_mix.has_gas(/datum/gas/oxygen)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_TOO MUCH_OXYGEN when it wasn't expeced.")
+	if(lab_rat.has_alert(ALERT_TOO_MUCH_NITRO) && (!test_lungs.safe_nitro_max || !test_mix.has_gas(/datum/gas/nitrogen)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_TOO_MUCH_NITROGEN when it wasn't expeced.")
+	if(lab_rat.has_alert(ALERT_TOO_MUCH_CO2) && (!test_lungs.safe_co2_max || !test_mix.has_gas(/datum/gas/carbon_dioxide)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_TOO_MUCH_CO2 when it wasn't expeced.")
+	if(lab_rat.has_alert(ALERT_TOO_MUCH_PLASMA) && (!test_lungs.safe_plasma_max || !test_mix.has_gas(/datum/gas/plasma)))
+		TEST_FAIL("Lungs check_breath() threw ALERT_TOO_MUCH_PLASMA when it wasn't expeced.")
+
+/// Tests the standard lungs organ to ensure it behaves the way we expect.
+/datum/unit_test/lungs_sanity/Run()
```

Comment:
Before defining any procs, have `/datum/unit_test/lungs_sanity` on its own. Move the documentation there.